package com.aditya.exception;

import java.time.LocalDateTime;

import org.springframework.http.HttpStatus;
import org.springframework.http.HttpStatusCode;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.ControllerAdvice;
import org.springframework.web.bind.annotation.ExceptionHandler;

import com.aditya.response.ErrorResponse;

@ControllerAdvice
public class GlobalExceptionHandler {

	@ExceptionHandler(PersonNotFoundException.class)
	public ResponseEntity<ErrorResponse> handlePersonNotFound(PersonNotFoundException ex)
	{
		return ResponseEntity<ErrorResponse>(new ErrorResponse("Person not found", ex.getMessage(), LocalDateTime.now()),HttpStatus.NOT_FOUND);
	}
}
